unseen/time_utils.py

```diff
@@ -217,35 +217,53 @@ def get_clim(ds, dims, time_period=None, groupby_init_month=False):
     return clim
 
 
-def select_time_period(ds, period):
+def select_time_period(da, period, time_name="time"):
     """Select a period of time.
 
-    Args:
-      ds (xarray DataSet or DataArray)
-      period (list) : Start and stop dates (in YYYY-MM-DD format)
-
-    Only works for cftime objects.
+    Parameters
+    ----------
+    da : xarray DataArray
+        Input array containing time dimension or variable. The time
+        dimension of variable should be cftime but can contain nans.
+    period : list of str
+        Start and stop dates (in YYYY-MM-DD format)
+    time_name: str
+        Name of the time dimension or variable
+
+    Returns
+    -------
+    masked : xarray DataArray
+        Array containing only times within provided period
```

Comment:
My general approach throughout the package has been to use Google style docstrings. We could swtich to Numpy style - we just need to be consistent across all modules.

https://sphinxcontrib-napoleon.readthedocs.io/en/latest/index.html#google-vs-numpy